lib/ansible/plugins/lookup/nios_next_network.py

```diff
@@ -0,0 +1,102 @@
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+lookup: nios_next_network
+version_added: ""
+short_description: Return the next available network for a network
+description:
+  - Uses the Infoblox WAPI API to return the next available network addresses
+    for a given network CIDR
+requirements:
+  - infoblox_client
+extends_documentation_fragment: nios
+options:
+    _terms:
+      description: The CIDR network to retrieve the next network from
+      required: True
+    cidr:
+      description: The CIDR of the network to retrieve the next network from
+      required: True
+    num:
+      description: The number of IP addresses to return
+      required: false
+      default: 1
+    exclude:
+      description: The number of IP addresses to return
+      required: false
+      default: ''
+"""
+
+EXAMPLES = """
+- name: return next available IP address for network 192.168.10.0/24
+  set_fact:
+    ipaddr: "{{ lookup('nios_next_network', '192.168.10.0/24', cidr=24, provider={'host': 'nios01', 'username': 'admin', 'password': 'password'}) }}"
+
+- name: return the next 3 available IP addresses for network 192.168.10.0/24
+  set_fact:
+    ipaddr: "{{ lookup('nios_next_network', '192.168.10.0/24', cidr=24, num=3, provider={'host': 'nios01', 'username': 'admin', 'password': 'password'}) }}"
+"""
+
+RETURN = """
+_list:
+  description:
+    - The list of next network addresses available
+  returned: always
+  type: list
+"""
+
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils.net_tools.nios.api import WapiLookup
+from ansible.module_utils._text import to_text
+from ansible.errors import AnsibleError
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, **kwargs):
+        try:
+            network = terms[0]
+        except IndexError:
+            raise AnsibleError('missing network argument in the form of A.B.C.D/E')
+        try:
+            cidr = kwargs.get('cidr', 24)
+        except IndexError:
+            raise AnsibleError('missing CIDR argument in the form of xx')
+
+        provider = kwargs.pop('provider', {})
+        wapi = WapiLookup(provider)
+
+        network_obj = wapi.get_object('networkcontainer', {'network': network})
```

Comment:
@dionben if the user will try to search for `network` made via normal `network` it will fail as this line will get the object only if the user has made a `network container` which is not generic, as Infoblox allows to search for next available network even in `network` objects along with `networkcontainer` objects. So, please include the check for checking `network` objects as well.